src/System/Random/Internal.hs

```diff
@@ -241,6 +245,25 @@ class Monad m => StatefulGen g m where
 
 
 
+-- | This class is designed for stateful RNGs that can be saved as immutable data type.
+--
+-- @since 1.2
+class StatefulGen (MutableGen f m) m => FrozenGen f m where
+  -- | Represents the state of the pseudo-random number generator for use with
+  -- 'thawGen' and 'freezeGen'.
+  --
+  -- @since 1.2
+  type MutableGen f m = (g :: Type) | g -> f
+  -- | Saves the state of the pseudo-random number generator as a frozen 'Seed'.
```

Comment:
```suggestion
  -- | Saves the state of the pseudo-random number generator as a frozen seed.
```

`'Seed'` would make this a broken link to a non-existing declaration of type or constructor called `Seed`.

Reply:
I initially though about calling it `Seed`, but the whole thing got flipped and the type family is called `Mutable`. But the seed and frozen are pretty synonymous in this context so this a perfect fix. Good catch